### ISSUE TITLE
Encryption for local storage

### ISSUE THREAD

**Issue by psgs** [Carlgo11/UploadMe#3]
Hello!

As I was browsing ```download.php```, I noticed a few things regarding encryption for storage that may improve UploadMe's system security.

At the top of ```download.php```, decryption seems to occur automatically, as shown in the following code block:
```php
function decrypt($encrypted, $password, $salt = '!kQm*fF3pXe1Kbm%9')
{
    $key = hash('SHA256', $salt . $password, true);
    $iv = base64_decode(substr($encrypted, 0, 22) . '==');
    $encrypted = substr($encrypted, 22);
    $decrypted = rtrim(mcrypt_decrypt(MCRYPT_RIJNDAEL_128, $key, base64_decode($encrypted), MCRYPT_MODE_CBC, $iv), "\0\4");
    $hash = substr($decrypted, -32);
    $decrypted = substr($decrypted, 0, -32);
    if (md5($decrypted) != $hash)
        return false;
    return $decrypted;
}
```

Similar logic can be seen when encrypting data:
```php
function encrypt($decrypted, $password, $salt = '!kQm*fF3pXe1Kbm%9')
{
    $key = hash('SHA256', $salt . $password, true);
    srand();
    $iv = mcrypt_create_iv(mcrypt_get_iv_size(MCRYPT_RIJNDAEL_128, MCRYPT_MODE_CBC), MCRYPT_RAND);
    if (strlen($iv_base64 = rtrim(base64_encode($iv), '=')) != 22)
        return false;
    $encrypted = base64_encode(mcrypt_encrypt(MCRYPT_RIJNDAEL_128, $key, $decrypted . md5($decrypted), MCRYPT_MODE_CBC, $iv));
    return $iv_base64 . $encrypted;
}
```

Are files effectively encrypted for local storage, even when users do not supply a password?

When encrypting or decrypting, a hard-coded salt is simply concatenated to a file's password. Would it improve the system's security to instead access a salt from a separate location, which may change from time to time?
Also, would mixing the salt with the password in a more complex fashion increase encryption strength?

Thankyou,
psgs :palm_tree:

**Comment by Carlgo11** [comment]
Hello psgs, since encryption() and decryption() are functions they're not run each time but only when called.

When it comes to salt that has been on my mind to fix. I'll probably do a few random bytes as a salt. The MySQL table already supports salt input. 

Thanks for bringing it up tho! I may or may not have forgotten about it *cough* :wink:

**Comment by Carlgo11** [comment]
Fixed in 08ea352
